{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "33184628-61b4-4687-976f-e929a84ef0e6",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Download gene annotations v42 (latest) - of CHR region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7b823264-e35d-41f8-a8f5-a395ff090a37",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2023-01-10 21:59:24--  https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/gencode.v42.annotation.gtf.gz\n",
      "Resolving ftp.ebi.ac.uk (ftp.ebi.ac.uk)... 193.62.193.138\n",
      "Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.193.138|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 49548290 (47M) [application/x-gzip]\n",
      "Saving to: ‘/s/project/first_last_exon/Data/input_data/gencode.v42.annotation.gtf.gz’\n",
      "\n",
      "/s/project/first_la 100%[===================>]  47.25M  1.85MB/s    in 23s     \n",
      "\n",
      "2023-01-10 21:59:48 (2.01 MB/s) - ‘/s/project/first_last_exon/Data/input_data/gencode.v42.annotation.gtf.gz’ saved [49548290/49548290]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget -O /s/project/first_last_exon/Data/input_data/gencode.v42.annotation.gtf.gz https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/gencode.v42.annotation.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c7dd1c04-9123-49b2-abbf-81d734f8440d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# unzip annotations file\n",
    "!gzip -d /s/project/first_last_exon/Data/input_data/gencode.v42.annotation.gtf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "514fc25b-699c-4aa3-b6bf-b51786847e31",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Step 1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1d1c9d6-5eef-44ec-9781-b9d271e2a290",
   "metadata": {},
   "source": [
    "### Run annotation script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "48af3cae-98f9-4237-8d4f-7cdbe7aeb5bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# see annotations file content\n",
    "\n",
    "import pyranges as pr\n",
    "\n",
    "# as PyRanges-object\n",
    "gr = pr.read_gtf(\"/s/project/first_last_exon/Data/input_data/gencode.v42.annotation.gtf\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5ab856a6-5a59-4893-9cb0-9e8802f95e2a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Source</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Frame</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>...</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>havana_transcript</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>exon_id</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>ont</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>ccdsid</th>\n",
       "      <th>artif_dupl</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11868</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000290825.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>11868</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000290825.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>OTTHUMT00000362751.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>11868</td>\n",
       "      <td>12227</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000290825.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>OTTHUMT00000362751.1</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00002234944.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>12612</td>\n",
       "      <td>12721</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000290825.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>OTTHUMT00000362751.1</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSE00003582793.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>13220</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000290825.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>OTTHUMT00000362751.1</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSE00002312635.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413836</th>\n",
       "      <td>chrY</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>57214349</td>\n",
       "      <td>57214397</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227159.8_PAR_Y</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>OTTHUMT00000058841.1</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSE00002072208.1</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>OTTHUMG00000022678.1</td>\n",
       "      <td>PGO:0000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413837</th>\n",
       "      <td>chrY</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>57213879</td>\n",
       "      <td>57213964</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227159.8_PAR_Y</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>OTTHUMT00000058841.1</td>\n",
       "      <td>2</td>\n",
       "      <td>ENSE00002046926.1</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>OTTHUMG00000022678.1</td>\n",
       "      <td>PGO:0000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413838</th>\n",
       "      <td>chrY</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>57213525</td>\n",
       "      <td>57213602</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227159.8_PAR_Y</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>OTTHUMT00000058841.1</td>\n",
       "      <td>3</td>\n",
       "      <td>ENSE00002021169.1</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>OTTHUMG00000022678.1</td>\n",
       "      <td>PGO:0000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413839</th>\n",
       "      <td>chrY</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>57213203</td>\n",
       "      <td>57213357</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227159.8_PAR_Y</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>OTTHUMT00000058841.1</td>\n",
       "      <td>4</td>\n",
       "      <td>ENSE00002036959.1</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>OTTHUMG00000022678.1</td>\n",
       "      <td>PGO:0000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413840</th>\n",
       "      <td>chrY</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>57212183</td>\n",
       "      <td>57213125</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227159.8_PAR_Y</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>OTTHUMT00000058841.1</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSE00002023900.1</td>\n",
       "      <td>HGNC:37115</td>\n",
       "      <td>OTTHUMG00000022678.1</td>\n",
       "      <td>PGO:0000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3413841 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "+--------------+------------+------------+-----------+-----------+-------+\n",
       "| Chromosome   | Source     | Feature    | Start     | End       | +21   |\n",
       "| (category)   | (object)   | (object)   | (int32)   | (int32)   | ...   |\n",
       "|--------------+------------+------------+-----------+-----------+-------|\n",
       "| chr1         | HAVANA     | gene       | 11868     | 14409     | ...   |\n",
       "| chr1         | HAVANA     | transcript | 11868     | 14409     | ...   |\n",
       "| chr1         | HAVANA     | exon       | 11868     | 12227     | ...   |\n",
       "| chr1         | HAVANA     | exon       | 12612     | 12721     | ...   |\n",
       "| ...          | ...        | ...        | ...       | ...       | ...   |\n",
       "| chrY         | HAVANA     | exon       | 57213879  | 57213964  | ...   |\n",
       "| chrY         | HAVANA     | exon       | 57213525  | 57213602  | ...   |\n",
       "| chrY         | HAVANA     | exon       | 57213203  | 57213357  | ...   |\n",
       "| chrY         | HAVANA     | exon       | 57212183  | 57213125  | ...   |\n",
       "+--------------+------------+------------+-----------+-----------+-------+\n",
       "Stranded PyRanges object has 3,413,841 rows and 26 columns from 25 chromosomes.\n",
       "For printing, the PyRanges was sorted on Chromosome and Strand.\n",
       "21 hidden columns: Score, Strand, Frame, gene_id, gene_type, gene_name, level, ... (+ 14 more.)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6d9a55f7-8989-4db5-9b3e-6b660c874aa3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex\n"
     ]
    }
   ],
   "source": [
    "%cd HITindex/hitindex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee44fef2-830f-4d54-bfc0-8da68ec41bda",
   "metadata": {},
   "outputs": [],
   "source": [
    "!python HITindex_annotate.py --gtf /s/project/first_last_exon/Data/input_data/gencode.v42.annotation.gtf --ss3buffer 50 --ss5buffer 20 --outfile /s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_1/metaexons.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d419e0c3-23ee-4d71-af8e-4dd45ffe44f1",
   "metadata": {},
   "source": [
    "Read output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "43b3caf4-c3cf-4c8e-b381-a9a65950fd08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output files\n",
      " ['metaexons.bed_ss3-50ss5-20.buffer', 'metaexons.bed', 'metaexons.bed_constituents']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os \n",
    "\n",
    "print(\"Output files\\n\", os.listdir(\"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_1/\"))\n",
    "\n",
    "file_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_1/metaexons.bed\"\n",
    "\n",
    "file_read = pd.read_csv(file_path, sep='\\t', names=('chr', 'start', 'stop', 'meta', 'count', 'strand' ), header=None, index_col=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b467c1ee-ebbc-4dfd-b508-5286b21939ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>meta</th>\n",
       "      <th>count</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>12227</td>\n",
       "      <td>chr1:11869-12227;ENSG00000290825.1;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12613</td>\n",
       "      <td>12721</td>\n",
       "      <td>chr1:12613-12721;ENSG00000290825.1;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13221</td>\n",
       "      <td>14409</td>\n",
       "      <td>chr1:13221-14409;ENSG00000290825.1;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12010</td>\n",
       "      <td>12057</td>\n",
       "      <td>chr1:12010-12057;ENSG00000223972.6;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>12179</td>\n",
       "      <td>12227</td>\n",
       "      <td>chr1:12179-12227;ENSG00000223972.6;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358272</th>\n",
       "      <td>chrM</td>\n",
       "      <td>14149</td>\n",
       "      <td>14673</td>\n",
       "      <td>chrM:14149-14673;ENSG00000198695.2;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358273</th>\n",
       "      <td>chrM</td>\n",
       "      <td>14674</td>\n",
       "      <td>14742</td>\n",
       "      <td>chrM:14674-14742;ENSG00000210194.1;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358274</th>\n",
       "      <td>chrM</td>\n",
       "      <td>14747</td>\n",
       "      <td>15887</td>\n",
       "      <td>chrM:14747-15887;ENSG00000198727.2;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358275</th>\n",
       "      <td>chrM</td>\n",
       "      <td>15888</td>\n",
       "      <td>15953</td>\n",
       "      <td>chrM:15888-15953;ENSG00000210195.2;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358276</th>\n",
       "      <td>chrM</td>\n",
       "      <td>15956</td>\n",
       "      <td>16023</td>\n",
       "      <td>chrM:15956-16023;ENSG00000210196.2;TXPT:1;FE:0...</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>358277 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr  start   stop                                               meta  \\\n",
       "0       chr1  11869  12227  chr1:11869-12227;ENSG00000290825.1;TXPT:1;FE:0...   \n",
       "1       chr1  12613  12721  chr1:12613-12721;ENSG00000290825.1;TXPT:1;FE:0...   \n",
       "2       chr1  13221  14409  chr1:13221-14409;ENSG00000290825.1;TXPT:1;FE:0...   \n",
       "3       chr1  12010  12057  chr1:12010-12057;ENSG00000223972.6;TXPT:1;FE:0...   \n",
       "4       chr1  12179  12227  chr1:12179-12227;ENSG00000223972.6;TXPT:1;FE:0...   \n",
       "...      ...    ...    ...                                                ...   \n",
       "358272  chrM  14149  14673  chrM:14149-14673;ENSG00000198695.2;TXPT:1;FE:0...   \n",
       "358273  chrM  14674  14742  chrM:14674-14742;ENSG00000210194.1;TXPT:1;FE:0...   \n",
       "358274  chrM  14747  15887  chrM:14747-15887;ENSG00000198727.2;TXPT:1;FE:0...   \n",
       "358275  chrM  15888  15953  chrM:15888-15953;ENSG00000210195.2;TXPT:1;FE:0...   \n",
       "358276  chrM  15956  16023  chrM:15956-16023;ENSG00000210196.2;TXPT:1;FE:0...   \n",
       "\n",
       "        count strand  \n",
       "0           0      +  \n",
       "1           0      +  \n",
       "2           0      +  \n",
       "3           0      +  \n",
       "4           0      +  \n",
       "...       ...    ...  \n",
       "358272      0      -  \n",
       "358273      0      -  \n",
       "358274      0      +  \n",
       "358275      0      +  \n",
       "358276      0      -  \n",
       "\n",
       "[358277 rows x 6 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_read"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "719aa850-b158-48d0-b1ec-95aa4f50e18c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Step 2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf8ae791-a0de-4688-858f-05fe24d3f7db",
   "metadata": {},
   "source": [
    "### Run classification script on all CHR bam files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "07a53018-1a17-4796-9f3d-28e85beefae8",
   "metadata": {},
   "outputs": [],
   "source": [
    "class args:\n",
    "    def __init__(self, junctionReads, bam, juncbam, readstrand, HITindex, bed, classify, calculatePSI, metricsID, outname):\n",
    "        self.junctionReads = junctionReads\n",
    "        self.bam = bam\n",
    "        self.juncbam = juncbam\n",
    "        self.readstrand = readstrand\n",
    "        self.HITindex = HITindex\n",
    "        self.bed = bed\n",
    "        self.classify = classify\n",
    "        self.calculatePSI = calculatePSI\n",
    "        self.metricsID = metricsID\n",
    "        self.outname = outname\n",
    "        self.readtype = 'paired'\n",
    "        self.overlap = 10\n",
    "        self.readnum = 2\n",
    "        self.parameters = \"HIT_identity_parameters.txt\"\n",
    "        self.bootstrap = 1000\n",
    "        self.edge  = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "41581873-60d1-44e2-9cb6-751b85969ccc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100 files detected\n",
      "Computing sample: NA20796.1.M_120209_2, 1/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/ouga/home/ag_gagneur/afi/.conda/envs/first_last_exon/lib/python3.6/site-packages/pymc3/data.py:307: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.\n",
      "  self.shared = theano.shared(data[in_memory_slc])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " |██████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 3,240.6]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,240.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py:505: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  afehere['sumR-L'] = sum(RLlist)\n",
      "/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py:506: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  afehere['AFEPSI'] = AFEPSI\n",
      "/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py:516: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  alehere['sumL-R'] = sum(LRlist)\n",
      "/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py:517: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  alehere['ALEPSI'] = ALEPSI\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... PSI calculated.\n",
      "Sample NA20796.1.M_120209_2 is done!\n",
      "Computing sample: NA20805.4.M_120208_7, 2/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |████████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 2,863]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 2,863\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample NA20805.4.M_120208_7 is done!\n",
      "Computing sample: HG00278.1.M_111124_5, 3/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:40<00:00 Average Loss = 2,825.3]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 2,825.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample HG00278.1.M_111124_5 is done!\n",
      "Computing sample: NA20539.4.M_120208_5, 4/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |████████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 2,864]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 2,864.2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample NA20539.4.M_120208_5 is done!\n",
      "Computing sample: NA20582.4.M_120208_2, 5/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:40<00:00 Average Loss = 3,046.5]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,046.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample NA20582.4.M_120208_2 is done!\n",
      "Computing sample: NA19207.4.M_120208_5, 6/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:38<00:00 Average Loss = 3,038.9]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,039\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample NA19207.4.M_120208_5 is done!\n",
      "Computing sample: HG00356.2.M_111215_6, 7/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 3,073.2]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,073.4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample HG00356.2.M_111215_6 is done!\n",
      "Computing sample: HG00344.4.M_120208_2, 8/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 2,933.9]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 2,934.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample HG00344.4.M_120208_2 is done!\n",
      "Computing sample: HG00116.2.M_120131_1, 9/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 3,071.5]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,071.8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample HG00116.2.M_120131_1 is done!\n",
      "Computing sample: NA18916.2.M_111215_6, 10/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 3,183.3]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,183.6\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample NA18916.2.M_111215_6 is done!\n",
      "Computing sample: HG00189.1.M_111124_5, 11/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:39<00:00 Average Loss = 2,966.2]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 2,966.3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n",
      "...calculated significance.\n",
      "... exons identified.\n",
      "Calculating PSI...\n",
      "... PSI calculated.\n",
      "Sample HG00189.1.M_111124_5 is done!\n",
      "Computing sample: NA11994.2.M_111216_7, 12/100\n",
      "Obtaining junctions...\n",
      "... junctions obtained.\n",
      "Calculating metric...\n",
      "... 358277 exons processed in 62696 genes.\n",
      "... got exons...\n",
      "NOTE: unstranded reads\n",
      "... got junctions...\n",
      "... got counts...\n",
      "HIT for exons...1000..2000..3000..4000..5000..6000..7000..8000..9000..10000..11000..12000..13000..14000..15000..16000..17000..18000..19000..20000..21000..22000..23000..24000..25000..26000..27000..28000..29000..30000..31000..32000..33000..34000..35000..36000..37000..38000..39000..40000..41000..42000..43000..44000..45000..46000..47000..48000..49000..50000..51000..52000..53000..54000..55000..56000..57000..58000..59000..60000..61000..62000..63000..64000..65000..66000..67000..68000..69000..70000..71000..72000..73000..74000..75000..76000..77000..78000..79000..80000..81000..82000..83000..84000..85000..86000..87000..88000..89000..90000..91000..92000..93000..94000..95000..96000..97000..98000..99000..100000..101000..102000..103000..104000..105000..106000..107000..108000..109000..110000..111000..112000..113000..114000..115000..116000..117000..118000..119000..120000..121000..122000..123000..124000..125000..126000..127000..128000..129000..130000..131000..132000..133000..134000..135000..136000..137000..138000..139000..140000..141000..142000..143000..144000..145000..146000..147000..148000..149000..150000..151000..152000..153000..154000..155000..156000..157000..158000..159000..160000..161000..162000..163000..164000..165000..166000..167000..168000..169000..170000..171000..172000..173000..174000..175000..176000..177000..178000..179000..180000..181000..182000..183000..184000..185000..186000..187000..188000..189000..190000..191000..192000..193000..194000..195000..196000..197000..198000..199000..200000..201000..202000..203000..204000..205000..206000..207000..208000..209000..210000..211000..212000..213000..214000..215000..216000..217000..218000..219000..220000..221000..222000..223000..224000..225000..226000..227000..228000..229000..230000..231000..232000..233000..234000..235000..236000..237000..238000..239000..240000..241000..242000..243000..244000..245000..246000..247000..248000..249000..250000..251000..252000..253000..254000..255000..256000..257000..258000..259000..260000..261000..262000..263000..264000..265000..266000..267000..268000..269000..270000..271000..272000..273000..274000..275000..276000..277000..278000..279000..280000..281000..282000..283000..284000..285000..286000..287000..288000..289000..290000..291000..292000..293000..294000..295000..296000..297000..298000..299000..300000..301000..302000..303000..304000..305000..306000..307000..308000..309000..310000..311000..312000..313000..314000..315000..316000..317000..318000..319000..320000..321000..322000..323000..324000..325000..326000..327000..328000..329000..330000..331000..332000..333000..334000..335000..336000..337000..338000..339000..340000..341000..342000..343000..344000..345000..346000..347000..348000..349000..350000..351000..352000..353000..354000..355000..356000..357000..358000..358277.\n",
      "... HITindex calculated.\n",
      "... edge effect exons flagged.\n",
      "... metrics written.\n",
      " |██████████████████████████████████████| 100.00% [100000/100000 01:40<00:00 Average Loss = 3,153.4]\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Finished [100%]: Average Loss = 3,153.5\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "... generative model calculated.\n",
      "Classifying exons...\n",
      "...read exons.\n",
      "...read parameters.\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-54-8d4b3da4cea3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     22\u001b[0m     \u001b[0marguments\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjunctionReads\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbam\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mjuncbam\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreadstrand\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mHITindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbed\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mclassify\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcalculatePSI\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmetricsID\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     23\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Computing sample: {}, {}/{}\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msample_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbam_files_list\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 24\u001b[0;31m     \u001b[0mhitmain\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marguments\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     25\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Sample {} is done!\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msample_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py\u001b[0m in \u001b[0;36mmain\u001b[0;34m(args)\u001b[0m\n\u001b[1;32m    607\u001b[0m                 \u001b[0mparamdict\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreadParameters\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparameters\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    608\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"...read parameters.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 609\u001b[0;31m                 \u001b[0mHITcombo\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msignificance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mHITcombo\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbootstrap\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfloat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mparamdict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'HIThybrid'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexon_outname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    610\u001b[0m                 \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"...calculated significance.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    611\u001b[0m                 \u001b[0mHITidentify\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcall_terminal\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mHITcombo\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparamdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexon_outname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py\u001b[0m in \u001b[0;36msignificance\u001b[0;34m(HITcombo, n_boot, cutoff, outname)\u001b[0m\n\u001b[1;32m    443\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    444\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0msignificance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mHITcombo\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_boot\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcutoff\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 445\u001b[0;31m         \u001b[0mHITcombo\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'CI_75'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'CI_90'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'CI_95'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'pval_CI'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'pval_internal'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mHITcombo\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'nUP'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'nDOWN'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'HITindex'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbootstrap\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult_type\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"expand\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_boot\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mn_boot\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcutoff\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcutoff\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    446\u001b[0m         \u001b[0;31m#HITcombo['pval_IntronNull'] = HITcombo[['nUP', 'nDOWN', 'HITindex']].apply(bootIntronNull, axis=1, n_boot=n_boot)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    447\u001b[0m         \u001b[0;31m#HITcombo[['CI_75', 'CI_90', 'CI_95', 'pval_CI']] = HITcombo[['nUP','nDOWN']].apply(bootCI, axis=1, result_type=\"expand\", n_boot=n_boot, cutoff=cutoff)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/ouga/home/ag_gagneur/afi/.conda/envs/first_last_exon/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, axis, broadcast, raw, reduce, result_type, args, **kwds)\u001b[0m\n\u001b[1;32m   6926\u001b[0m             \u001b[0mkwds\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6927\u001b[0m         )\n\u001b[0;32m-> 6928\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   6929\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6930\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapplymap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/ouga/home/ag_gagneur/afi/.conda/envs/first_last_exon/lib/python3.6/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    184\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    185\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 186\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    187\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    188\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapply_empty_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/ouga/home/ag_gagneur/afi/.conda/envs/first_last_exon/lib/python3.6/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    290\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    291\u001b[0m         \u001b[0;31m# compute the result using the series generator\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 292\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_series_generator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    293\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    294\u001b[0m         \u001b[0;31m# wrap results\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/ouga/home/ag_gagneur/afi/.conda/envs/first_last_exon/lib/python3.6/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_series_generator\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    319\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    320\u001b[0m                 \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mv\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries_gen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 321\u001b[0;31m                     \u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    322\u001b[0m                     \u001b[0mkeys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    323\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/ouga/home/ag_gagneur/afi/.conda/envs/first_last_exon/lib/python3.6/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mf\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m    110\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    111\u001b[0m             \u001b[0;32mdef\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 112\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    113\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    114\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py\u001b[0m in \u001b[0;36mbootstrap\u001b[0;34m(exon, n_boot, cutoff)\u001b[0m\n\u001b[1;32m    420\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mboot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_boot\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    421\u001b[0m                 \u001b[0mbootHITs_internal\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msampleReads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mallreads_internal\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnBOTH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 422\u001b[0;31m                 \u001b[0mbootHITs_CI\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msampleReads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mallreads_CI\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnBOTH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    423\u001b[0m         \u001b[0mbootHITs_internal\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbootHITs_internal\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    424\u001b[0m         \u001b[0mbootHITs_CI\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbootHITs_CI\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/nasif12/home_if12/afi/first_last_exon/HITindex/hitindex/HITindex_classify_mainfn.py\u001b[0m in \u001b[0;36msampleReads\u001b[0;34m(allreads, nBOTH)\u001b[0m\n\u001b[1;32m    394\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    395\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0msampleReads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mallreads\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnBOTH\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 396\u001b[0;31m         \u001b[0mnchoice\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrandom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchoice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mallreads\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnBOTH\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreplace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    397\u001b[0m         \u001b[0mnchoice_up\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnchoice\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    398\u001b[0m         \u001b[0mnchoice_down\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnchoice\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mmtrand.pyx\u001b[0m in \u001b[0;36mnumpy.random.mtrand.RandomState.choice\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m<__array_function__ internals>\u001b[0m in \u001b[0;36mprod\u001b[0;34m(*args, **kwargs)\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import HITindex_classify_mainfn as hitmain\n",
    "\n",
    "datafolder_path = \"/s/project/first_last_exon/Data/input_data/rna-seq/\"\n",
    "output_folder_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/\"\n",
    "bed_file_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_1/metaexons.bed_ss3-50ss5-20.buffer\"\n",
    "bam_files_list = [f for f in os.listdir(datafolder_path) if f.endswith(\"_chr1.bam\")]\n",
    "print(len(bam_files_list), \"files detected\")\n",
    "\n",
    "for i, f in enumerate(bam_files_list):\n",
    "    sample_name = f.split(\"_chr1.bam\")[0]\n",
    "    junctionReads = True\n",
    "    bam = os.path.join(datafolder_path, f)\n",
    "    juncbam = os.path.join(output_folder_path, sample_name + \".junctions.bam\") \n",
    "    readstrand = \"fr-unstrand\"\n",
    "    HITindex = True\n",
    "    bed = bed_file_path \n",
    "    classify = True\n",
    "    calculatePSI = True\n",
    "    outname = os.path.join(output_folder_path, sample_name + \"_HITindex\")\n",
    "    arguments = args(junctionReads, bam, juncbam, readstrand, HITindex, bed, classify, calculatePSI, metricsID, outname)\n",
    "    print(\"Computing sample: {}, {}/{}\".format(sample_name, i+1, len(bam_files_list)))\n",
    "    hitmain.main(arguments)\n",
    "    print(\"Sample {} is done!\".format(sample_name))\n",
    "          \n",
    "\n",
    "# !python HITindex_classify.py --junctionReads --bam /s/project/first_last_exon/Data/input_data/rna-seq/HG00096.1.M_111124_6_chr1.bam --juncbam /s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1.junctions.bam --readstrand fr-unstrand --HITindex --bed /s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_1/metaexons.bed_ss3-50ss5-20.buffer --classify --calculatePSI --metricsID sampleHITindex.exon --outname /s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1_HITindex   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "4377a0d5-13a7-4765-9470-9848a82d1b3d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CompletedProcess(args='ls', returncode=0)\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a8e8a79d-921a-46a0-bca4-1996268fa9ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['@HD\\tVN:1.3\\n', '@SQ\\tSN:chr1\\tLN:249250621\\n', '@SQ\\tSN:chr10\\tLN:135534747\\n', '@SQ\\tSN:chr11\\tLN:135006516\\n', '@SQ\\tSN:chr12\\tLN:133851895\\n', '@SQ\\tSN:chr13\\tLN:115169878\\n', '@SQ\\tSN:chr14\\tLN:107349540\\n', '@SQ\\tSN:chr15\\tLN:102531392\\n', '@SQ\\tSN:chr16\\tLN:90354753\\n', '@SQ\\tSN:chr17\\tLN:81195210\\n', '@SQ\\tSN:chr18\\tLN:78077248\\n', '@SQ\\tSN:chr19\\tLN:59128983\\n', '@SQ\\tSN:chr2\\tLN:243199373\\n', '@SQ\\tSN:chr20\\tLN:63025520\\n', '@SQ\\tSN:chr21\\tLN:48129895\\n', '@SQ\\tSN:chr22\\tLN:51304566\\n', '@SQ\\tSN:chr3\\tLN:198022430\\n', '@SQ\\tSN:chr4\\tLN:191154276\\n', '@SQ\\tSN:chr5\\tLN:180915260\\n', '@SQ\\tSN:chr6\\tLN:171115067\\n', '@SQ\\tSN:chr7\\tLN:159138663\\n', '@SQ\\tSN:chr8\\tLN:146364022\\n', '@SQ\\tSN:chr9\\tLN:141213431\\n', '@SQ\\tSN:chrX\\tLN:155270560\\n', '@SQ\\tSN:chrY\\tLN:59373566\\n', '@RG\\tID:0\\tPG:GEM\\tPL:ILLUMINA\\tSM:0\\n', '@PG\\tID:GEM\\tPN:gem-2-sam\\tVN:1.306\\n']\n"
     ]
    }
   ],
   "source": [
    "# Read junction bam file\n",
    "import pysam\n",
    "\n",
    "samfile = pysam.AlignmentFile(\"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1.junctions_read2.bam\", \"rb\")\n",
    "for read in samfile.fetch('chr1', 100, 120):\n",
    "    print(read)\n",
    "\n",
    "samfile.close()\n",
    "\n",
    "\n",
    "\n",
    "header_file_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1.junctions.bam_header.txt\"\n",
    "with open(header_file_path, \"r\") as f:\n",
    "    lines = f.readlines()\n",
    "    print(lines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "5c4474ca-54b7-4ea3-ab63-ef7507c5efea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read output files\n",
    "import pandas as pd\n",
    "exon_file_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1_HITindex.exon\"\n",
    "AFEPSI_file_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1_HITindex.AFEPSI\"\n",
    "ALEPSI_file_path = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/HG00096.1.M_111124_6_chr1_HITindex.ALEPSI\"\n",
    "exon_df = pd.read_csv(exon_file_path, sep='\\t')\n",
    "AFEPSI_df = pd.read_csv(AFEPSI_file_path, sep='\\t')\n",
    "ALEPSI_df = pd.read_csv(ALEPSI_file_path, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a0361a26-3287-453c-8c97-29df6168d01d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon</th>\n",
       "      <th>gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>nTXPT</th>\n",
       "      <th>nFE</th>\n",
       "      <th>nINTERNAL</th>\n",
       "      <th>nLE</th>\n",
       "      <th>nSINGLE</th>\n",
       "      <th>nUP</th>\n",
       "      <th>nDOWN</th>\n",
       "      <th>...</th>\n",
       "      <th>PofIL</th>\n",
       "      <th>downstream_fraction</th>\n",
       "      <th>HIT_postmean</th>\n",
       "      <th>CI_75</th>\n",
       "      <th>CI_90</th>\n",
       "      <th>CI_95</th>\n",
       "      <th>pval_CI</th>\n",
       "      <th>pval_internal</th>\n",
       "      <th>ID</th>\n",
       "      <th>ID_position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:15005-15038</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>...</td>\n",
       "      <td>0.09185</td>\n",
       "      <td>0.49122</td>\n",
       "      <td>0.01756</td>\n",
       "      <td>-0.158,0.368</td>\n",
       "      <td>-0.263,0.474</td>\n",
       "      <td>-0.263,0.579</td>\n",
       "      <td>0.773</td>\n",
       "      <td>0.322</td>\n",
       "      <td>internal</td>\n",
       "      <td>last</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:15796-15947</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00012</td>\n",
       "      <td>0.77439</td>\n",
       "      <td>-0.54878</td>\n",
       "      <td>-0.8,-0.467</td>\n",
       "      <td>-0.867,-0.4</td>\n",
       "      <td>-0.867,-0.333</td>\n",
       "      <td>0.015</td>\n",
       "      <td>0.000</td>\n",
       "      <td>FirstInternal_high</td>\n",
       "      <td>FirstInternal_high</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:16607-16765</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00104</td>\n",
       "      <td>0.95758</td>\n",
       "      <td>-0.91515</td>\n",
       "      <td>-1.0,-0.333</td>\n",
       "      <td>-1.0,-0.333</td>\n",
       "      <td>-1.0,-0.325</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.017</td>\n",
       "      <td>first</td>\n",
       "      <td>first</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:16858-17055</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08771</td>\n",
       "      <td>0.01269</td>\n",
       "      <td>0.97463</td>\n",
       "      <td>0.556,1.0</td>\n",
       "      <td>0.556,1.0</td>\n",
       "      <td>0.333,1.0</td>\n",
       "      <td>0.006</td>\n",
       "      <td>0.004</td>\n",
       "      <td>last</td>\n",
       "      <td>last</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:17233-17368</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99959</td>\n",
       "      <td>0.10753</td>\n",
       "      <td>0.78493</td>\n",
       "      <td>0.688,0.875</td>\n",
       "      <td>0.656,0.906</td>\n",
       "      <td>0.625,0.938</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>InternalLast_high</td>\n",
       "      <td>InternalLast_high</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               exon               gene strand  nTXPT  nFE  nINTERNAL  nLE  \\\n",
       "0  chr1:15005-15038  ENSG00000227232.5      -      1    0          0    0   \n",
       "1  chr1:15796-15947  ENSG00000227232.5      -      1    0          0    0   \n",
       "2  chr1:16607-16765  ENSG00000227232.5      -      1    0          0    0   \n",
       "3  chr1:16858-17055  ENSG00000227232.5      -      1    0          0    0   \n",
       "4  chr1:17233-17368  ENSG00000227232.5      -      1    0          0    0   \n",
       "\n",
       "   nSINGLE  nUP  nDOWN  ...    PofIL  downstream_fraction  HIT_postmean  \\\n",
       "0        0   11      8  ...  0.09185              0.49122       0.01756   \n",
       "1        0    5     25  ...  0.00012              0.77439      -0.54878   \n",
       "2        0    0      6  ...  0.00104              0.95758      -0.91515   \n",
       "3        0    9      0  ...  0.08771              0.01269       0.97463   \n",
       "4        0   58      6  ...  0.99959              0.10753       0.78493   \n",
       "\n",
       "          CI_75         CI_90          CI_95  pval_CI  pval_internal  \\\n",
       "0  -0.158,0.368  -0.263,0.474   -0.263,0.579    0.773          0.322   \n",
       "1   -0.8,-0.467   -0.867,-0.4  -0.867,-0.333    0.015          0.000   \n",
       "2   -1.0,-0.333   -1.0,-0.333    -1.0,-0.325    0.023          0.017   \n",
       "3     0.556,1.0     0.556,1.0      0.333,1.0    0.006          0.004   \n",
       "4   0.688,0.875   0.656,0.906    0.625,0.938    0.000          0.000   \n",
       "\n",
       "                   ID         ID_position  \n",
       "0            internal                last  \n",
       "1  FirstInternal_high  FirstInternal_high  \n",
       "2               first               first  \n",
       "3                last                last  \n",
       "4   InternalLast_high   InternalLast_high  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "dfbbd984-a359-4715-9707-ed4cb143bffc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon</th>\n",
       "      <th>gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>nTXPT</th>\n",
       "      <th>nFE</th>\n",
       "      <th>nINTERNAL</th>\n",
       "      <th>nLE</th>\n",
       "      <th>nSINGLE</th>\n",
       "      <th>nUP</th>\n",
       "      <th>nDOWN</th>\n",
       "      <th>HITindex</th>\n",
       "      <th>dist_to_TSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:15005-15038</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>0.158</td>\n",
       "      <td>9853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:15796-15947</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>25</td>\n",
       "      <td>-0.667</td>\n",
       "      <td>8944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:16607-16765</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>8126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:16858-17055</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>7836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:17233-17368</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>6</td>\n",
       "      <td>0.812</td>\n",
       "      <td>7523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1261</th>\n",
       "      <td>chr1:247353953-247354287</td>\n",
       "      <td>ENSG00000289234.2</td>\n",
       "      <td>+</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1262</th>\n",
       "      <td>chr1:247374007-247374143</td>\n",
       "      <td>ENSG00000289234.2</td>\n",
       "      <td>+</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>20054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1263</th>\n",
       "      <td>chr1:247374007-247374143</td>\n",
       "      <td>ENSG00000289728.1</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1264</th>\n",
       "      <td>chr1:247491068-247492415</td>\n",
       "      <td>ENSG00000290684.1</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>-0.176</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1265</th>\n",
       "      <td>chr1:247491128-247492090</td>\n",
       "      <td>ENSG00000203664.8</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1266 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          exon               gene strand  nTXPT  nFE  \\\n",
       "0             chr1:15005-15038  ENSG00000227232.5      -      1    0   \n",
       "1             chr1:15796-15947  ENSG00000227232.5      -      1    0   \n",
       "2             chr1:16607-16765  ENSG00000227232.5      -      1    0   \n",
       "3             chr1:16858-17055  ENSG00000227232.5      -      1    0   \n",
       "4             chr1:17233-17368  ENSG00000227232.5      -      1    0   \n",
       "...                        ...                ...    ...    ...  ...   \n",
       "1261  chr1:247353953-247354287  ENSG00000289234.2      +     16    0   \n",
       "1262  chr1:247374007-247374143  ENSG00000289234.2      +     16    0   \n",
       "1263  chr1:247374007-247374143  ENSG00000289728.1      +      1    0   \n",
       "1264  chr1:247491068-247492415  ENSG00000290684.1      +      2    0   \n",
       "1265  chr1:247491128-247492090  ENSG00000203664.8      +      1    0   \n",
       "\n",
       "      nINTERNAL  nLE  nSINGLE  nUP  nDOWN  HITindex  dist_to_TSS  \n",
       "0             0    0        0   11      8     0.158         9853  \n",
       "1             0    0        0    5     25    -0.667         8944  \n",
       "2             0    0        0    0      6    -1.000         8126  \n",
       "3             0    0        0    9      0     1.000         7836  \n",
       "4             0    0        0   58      6     0.812         7523  \n",
       "...         ...  ...      ...  ...    ...       ...          ...  \n",
       "1261          0    0        0    0     24    -1.000            0  \n",
       "1262          0    0        0    9      0     1.000        20054  \n",
       "1263          0    0        0    9      0     1.000            0  \n",
       "1264          0    0        0    7     10    -0.176            0  \n",
       "1265          0    0        0    7      0     1.000            0  \n",
       "\n",
       "[1266 rows x 12 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_df.iloc[:,:12]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "40fa9581-6f26-43b6-a3aa-82c39f11147b",
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_df.to_csv( \"./exon.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f9d7a66f-0338-477d-a180-99ccf7803dad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>exon</th>\n",
       "      <th>strand</th>\n",
       "      <th>nTXPT</th>\n",
       "      <th>nFE</th>\n",
       "      <th>nUP</th>\n",
       "      <th>nDOWN</th>\n",
       "      <th>HITindex</th>\n",
       "      <th>sumR-L</th>\n",
       "      <th>AFEPSI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>chr1:15796-15947</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>25</td>\n",
       "      <td>-0.667</td>\n",
       "      <td>86</td>\n",
       "      <td>0.232558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>chr1:16607-16765</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>86</td>\n",
       "      <td>0.069767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>chr1:17606-17742</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>60</td>\n",
       "      <td>-0.818</td>\n",
       "      <td>86</td>\n",
       "      <td>0.627907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>chr1:17915-18061</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>86</td>\n",
       "      <td>0.069767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000187642.10</td>\n",
       "      <td>chr1:976499-976641</td>\n",
       "      <td>-</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>-1.000</td>\n",
       "      <td>19</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 gene                exon strand  nTXPT  nFE  nUP  nDOWN  \\\n",
       "0   ENSG00000227232.5    chr1:15796-15947      -      1    0    5     25   \n",
       "1   ENSG00000227232.5    chr1:16607-16765      -      1    0    0      6   \n",
       "2   ENSG00000227232.5    chr1:17606-17742      -      1    0    6     60   \n",
       "3   ENSG00000227232.5    chr1:17915-18061      -      1    0    0      6   \n",
       "4  ENSG00000187642.10  chr1:976499-976641      -      4    0    0     19   \n",
       "\n",
       "   HITindex  sumR-L    AFEPSI  \n",
       "0    -0.667      86  0.232558  \n",
       "1    -1.000      86  0.069767  \n",
       "2    -0.818      86  0.627907  \n",
       "3    -1.000      86  0.069767  \n",
       "4    -1.000      19  1.000000  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AFEPSI_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1d6926e-c221-4c79-bfe3-131c8ee7694e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.conda-first_last_exon]",
   "language": "python",
   "name": "conda-env-.conda-first_last_exon-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
