{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "330c1818-2ff9-42a2-bf7d-11a0c56ed01a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We have 100 different .exon files\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "exon_files_dir = \"/s/project/first_last_exon/Data/output_data/HITindex/rna_seq/step_2/chr_files\"\n",
    "exon_files = [f for f in os.listdir(exon_files_dir) if f.endswith(\".exon\")]\n",
    "print(\"We have {} different .exon files\".format(len(exon_files)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "782c5e3d-5223-406f-8e4b-0a3b954502c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon</th>\n",
       "      <th>gene</th>\n",
       "      <th>strand</th>\n",
       "      <th>nTXPT</th>\n",
       "      <th>nFE</th>\n",
       "      <th>nINTERNAL</th>\n",
       "      <th>nLE</th>\n",
       "      <th>nSINGLE</th>\n",
       "      <th>nUP</th>\n",
       "      <th>nDOWN</th>\n",
       "      <th>...</th>\n",
       "      <th>PofIL</th>\n",
       "      <th>downstream_fraction</th>\n",
       "      <th>HIT_postmean</th>\n",
       "      <th>CI_75</th>\n",
       "      <th>CI_90</th>\n",
       "      <th>CI_95</th>\n",
       "      <th>pval_CI</th>\n",
       "      <th>pval_internal</th>\n",
       "      <th>ID</th>\n",
       "      <th>ID_position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1:15005-15038</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08036</td>\n",
       "      <td>0.49515</td>\n",
       "      <td>0.00969</td>\n",
       "      <td>-0.091,0.273</td>\n",
       "      <td>-0.136,0.364</td>\n",
       "      <td>-0.182,0.364</td>\n",
       "      <td>0.892</td>\n",
       "      <td>0.321</td>\n",
       "      <td>internal</td>\n",
       "      <td>last</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1:15796-15947</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.91973</td>\n",
       "      <td>-0.83947</td>\n",
       "      <td>-0.934,-0.77</td>\n",
       "      <td>-0.934,-0.705</td>\n",
       "      <td>-0.967,-0.705</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>FirstInternal_high</td>\n",
       "      <td>FirstInternal_high</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:16607-16765</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03351</td>\n",
       "      <td>0.56797</td>\n",
       "      <td>-0.13594</td>\n",
       "      <td>-1.0,0.2</td>\n",
       "      <td>-1.0,0.2</td>\n",
       "      <td>-1.0,0.6</td>\n",
       "      <td>0.412</td>\n",
       "      <td>0.216</td>\n",
       "      <td>internal</td>\n",
       "      <td>internal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:16858-17055</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.76961</td>\n",
       "      <td>0.18971</td>\n",
       "      <td>0.62059</td>\n",
       "      <td>0.5,1.0</td>\n",
       "      <td>0.333,1.0</td>\n",
       "      <td>0.333,1.0</td>\n",
       "      <td>0.020</td>\n",
       "      <td>0.007</td>\n",
       "      <td>InternalLast_medium</td>\n",
       "      <td>InternalLast_medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1:17233-17368</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>9</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99994</td>\n",
       "      <td>0.09892</td>\n",
       "      <td>0.80216</td>\n",
       "      <td>0.74,0.88</td>\n",
       "      <td>0.7,0.9</td>\n",
       "      <td>0.68,0.92</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>InternalLast_high</td>\n",
       "      <td>InternalLast_high</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               exon               gene strand  nTXPT  nFE  nINTERNAL  nLE  \\\n",
       "0  chr1:15005-15038  ENSG00000227232.5      -      1    0          0    0   \n",
       "1  chr1:15796-15947  ENSG00000227232.5      -      1    0          0    0   \n",
       "2  chr1:16607-16765  ENSG00000227232.5      -      1    0          0    0   \n",
       "3  chr1:16858-17055  ENSG00000227232.5      -      1    0          0    0   \n",
       "4  chr1:17233-17368  ENSG00000227232.5      -      1    0          0    0   \n",
       "\n",
       "   nSINGLE  nUP  nDOWN  ...    PofIL  downstream_fraction  HIT_postmean  \\\n",
       "0        0   24     20  ...  0.08036              0.49515       0.00969   \n",
       "1        0    4     57  ...  0.00000              0.91973      -0.83947   \n",
       "2        0    1      4  ...  0.03351              0.56797      -0.13594   \n",
       "3        0   11      1  ...  0.76961              0.18971       0.62059   \n",
       "4        0   91      9  ...  0.99994              0.09892       0.80216   \n",
       "\n",
       "          CI_75          CI_90          CI_95  pval_CI  pval_internal  \\\n",
       "0  -0.091,0.273   -0.136,0.364   -0.182,0.364    0.892          0.321   \n",
       "1  -0.934,-0.77  -0.934,-0.705  -0.967,-0.705    0.000          0.000   \n",
       "2      -1.0,0.2       -1.0,0.2       -1.0,0.6    0.412          0.216   \n",
       "3       0.5,1.0      0.333,1.0      0.333,1.0    0.020          0.007   \n",
       "4     0.74,0.88        0.7,0.9      0.68,0.92    0.000          0.000   \n",
       "\n",
       "                    ID          ID_position  \n",
       "0             internal                 last  \n",
       "1   FirstInternal_high   FirstInternal_high  \n",
       "2             internal             internal  \n",
       "3  InternalLast_medium  InternalLast_medium  \n",
       "4    InternalLast_high    InternalLast_high  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read exon files into dataframes\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "dataframes_ls = []\n",
    "for f in exon_files:\n",
    "    df = pd.read_csv(os.path.join(exon_files_dir, f),  sep='\\t')\n",
    "    dataframes_ls.append(df)\n",
    "    \n",
    "merged_df = pd.concat(dataframes_ls, sort=False)\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "996094cf-27fa-4cf7-9042-2755ed493234",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of unique genes = 1177\n"
     ]
    }
   ],
   "source": [
    "genes_set = merged_df[\"gene\"].unique()\n",
    "classes_labels_set = [\"first\", \"FirstInternal_medium\", \"FirstInternal_high\", \"last\", \"InternalLast_medium\", \"InternalLast_high\", \"internal\"]\n",
    "\n",
    "print(\"Total number of unique genes = {}\".format(len(genes_set)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e54c9053-1827-4096-a6ce-076885b1a962",
   "metadata": {},
   "outputs": [],
   "source": [
    "# seqname - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. Important note: the seqname must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.\n",
    "# source - name of the program that generated this feature, or the data source (database or project name)\n",
    "# feature - feature type name, e.g. Gene, Variation, Similarity\n",
    "# start - Start position* of the feature, with sequence numbering starting at 1.\n",
    "# end - End position* of the feature, with sequence numbering starting at 1.\n",
    "# score - A floating point value.\n",
    "# strand - defined as + (forward) or - (reverse).\n",
    "# frame - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..\n",
    "# attribute - A semicolon-separated list of tag-value pairs, providing additional information about each feature."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7b37a893-b66d-465e-ae6e-420b8fcb6463",
   "metadata": {},
   "outputs": [],
   "source": [
    "first_exons_merge_output = {\"seqname\": [],\n",
    "               \"source\": [],\n",
    "                \"feature\": [],\n",
    "                \"start\": [],\n",
    "                \"end\": [],\n",
    "                \"score\": [],\n",
    "                \"strand\": [],\n",
    "                \"frame\": [],\n",
    "                \"attribute\": []\n",
    "                           }\n",
    "last_exons_merge_output = {\"seqname\": [],\n",
    "               \"source\": [],\n",
    "                \"feature\": [],\n",
    "                \"start\": [],\n",
    "                \"end\": [],\n",
    "                \"score\": [],\n",
    "                \"strand\": [],\n",
    "                \"frame\": [],\n",
    "                \"attribute\": []\n",
    "                          }\n",
    "def add_entry(dic, gene, exon, strand):\n",
    "#     chr1:s-e\n",
    "    dic[\"seqname\"].append(\"chr1\")\n",
    "    dic[\"source\"].append(\"hitindex\")\n",
    "    dic[\"feature\"].append(\"exon\")\n",
    "    start, end = exon.split(\":\")[1].split(\"-\")\n",
    "    dic[\"start\"].append(start)\n",
    "    dic[\"end\"].append(end)\n",
    "    dic[\"score\"].append(\".\")\n",
    "    dic[\"strand\"].append(strand)\n",
    "    dic[\"frame\"].append(\".\")\n",
    "    dic[\"attribute\"].append(\"gene_id:\"+ gene)\n",
    "\n",
    "    \n",
    "# Extract first and last exons records\n",
    "for gene in genes_set:\n",
    "    gene_rows = merged_df[merged_df[\"gene\"] == gene]\n",
    "    exons_set = gene_rows[\"exon\"].unique()\n",
    "    for exon in exons_set:\n",
    "        labels_set = gene_rows[gene_rows[\"exon\"] == exon][\"ID_position\"].unique()\n",
    "        first = False\n",
    "        last = False\n",
    "        for l in labels_set:\n",
    "            if \"first\" in l.lower():\n",
    "                first = True\n",
    "            elif \"last\" in l.lower():\n",
    "                last = True\n",
    "        if first:\n",
    "            add_entry(first_exons_merge_output, gene, exon, gene_rows[\"strand\"].unique()[0])\n",
    "        if last:\n",
    "            add_entry(last_exons_merge_output, gene, exon, gene_rows[\"strand\"].unique()[0])\n",
    "            \n",
    "#         hybird = not (len(labels_set) == 1 and ((\"last\" in labels_set or \"first\" in labels_set or \"internal\" in labels_set)))\n",
    "#         exon_type = \"hybird\" if hybird else labels_set[0]\n",
    "#         add_entry(merge_output, gene, exon, gene_rows[\"strand\"].unique()[0], exon_type)\n",
    "first_exons_merge_output_df = pd.DataFrame(first_exons_merge_output)\n",
    "first_exons_merge_output_df.to_csv(\"first_exons_merge_output.gtf\", index=False, sep='\\t', header=False)\n",
    "\n",
    "last_exons_merge_output_df = pd.DataFrame(last_exons_merge_output)\n",
    "last_exons_merge_output_df.to_csv(\"last_exons_merge_output.gtf\", index=False, sep='\\t', header=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:.conda-first_last_exon]",
   "language": "python",
   "name": "conda-env-.conda-first_last_exon-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
